src/interceptors/fetch/index.ts

```diff
@@ -88,18 +88,26 @@ export class FetchInterceptor extends Interceptor<HttpRequestEventMap> {
         )
       }
 
-      const respondWith = (response: Response): Response => {
-        // Clone the mocked response for the "response" event listener.
-        // This way, the listener can read the response and not lock its body
-        // for the actual fetch consumer.
-        const responseClone = response.clone()
-
-        this.emitter.emit('response', {
-          response: responseClone,
-          isMockedResponse: true,
-          request: interactiveRequest,
-          requestId,
-        })
+      const responsePromise = new DeferredPromise<Response>()
```

Comment:
I had to rewrite the Fetch interceptor to return a deferred Promise so we can expose functions like `respondWith()` and `errorWith()` to asynchronous contexts (the `unhandledException` listener). The listener cannot "return" anything to the pending Fetch promise, so instead it controls the deferred promise, which is always returned.